src/components/FileDropZone/FileDropZone.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+
+import type {IconData} from '../..';
+import type {UseDropZoneAccept} from '../../hooks/useDropZone';
+import type {BaseInputControlProps} from '../controls/types';
+
+import {FileDropZoneProvider} from './FileDropZone.Provider';
+import {cnFileDropZone} from './FileDropZone.classname';
+import {FileDropZoneButton} from './parts/FileDropZone.Button';
+import {FileDropZoneDescription} from './parts/FileDropZone.Description';
+import {FileDropZoneIcon} from './parts/FileDropZone.Icon';
+import {FileDropZoneTitle} from './parts/FileDropZone.Title';
+
+import './FileDropZone.scss';
+
+export interface FileDropZoneProps extends Pick<BaseInputControlProps, 'validationState'> {
+    accept: UseDropZoneAccept;
+    onAdd: (files: File[]) => void;
```

Comment:
We usually use the name `onUpdate` for such callbacks, this is a familiar pattern for users of our library